File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
import sys
input=sys.stdin.readline
from collections import deque

sys.setrecursionlimit(111111)

def dfs(x):
    print(x, end=' ')
    vis[x]=True
    for i in range(1, n+1):
        if not vis[i] and g[x][i]==1:
            dfs(i)

def bfs(x):
    q=deque()
    q.append(x)
    vis[x]=False
    while q:
        y=q.popleft()
        print(y, end=' ')
        for i in range(1, n+1):
            if vis[i] and g[y][i]==1:
                q.append(i)
                vis[i]=False

n, m, v=map(int, input().split())
g=[[0]*(n+1) for _ in range(n+1)]
for i in range(m):
    x, y = map(int, input().split())
    g[y][x]=g[x][y]=1
vis=[False]*(n+1)
dfs(v)
print()
bfs(v)
# print(' '.join(list(map(str, dres))))
# print(' '.join(list(map(str, bres))))

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayDeque;
import java.util.ArrayList;
import java.util.Collections;
import java.util.Deque;
import java.util.List;
import java.util.StringTokenizer;

public class Main {
	static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
	static StringTokenizer st;
	static StringBuilder sb = new StringBuilder();
	static int n, m, v;
	static List<Integer>[] g;
	static boolean[] vis;

	public static void main(String[] args) throws IOException {
		st = new StringTokenizer(br.readLine());
		n = Integer.parseInt(st.nextToken());
		m = Integer.parseInt(st.nextToken());
		v = Integer.parseInt(st.nextToken());
		g = new ArrayList[n + 1];
		vis = new boolean[n + 1];
		for (int i = 1; i <= n; i++) {
			g[i] = new ArrayList<Integer>();
		}
		for (int i = 0; i < m; i++) {
			st = new StringTokenizer(br.readLine());
			int a = Integer.parseInt(st.nextToken());
			int b = Integer.parseInt(st.nextToken());
			g[a].add(b);
			g[b].add(a);
		}
		for (int i = 1; i <= n; i++) {
			Collections.sort(g[i]);
		}
		dfs(v);
		sb.append("\n");
		bfs();
		System.out.println(sb);
	}

	static void bfs() {
		boolean[] vis = new boolean[n + 1];
		Deque<Integer> q = new ArrayDeque<Integer>();
		q.offer(v);
		vis[v] = true;
		sb.append(v + " ");
		while (!q.isEmpty()) {
			int node = q.poll();
			for (int n : g[node]) {
				if (!vis[n]) {
					q.offer(n);
					vis[n] = true;
					sb.append(n + " ");
				}
			}
		}
	}

	static void dfs(int node) {
		vis[node] = true;
		sb.append(node + " ");
		for (int n : g[node]) {
			if (!vis[n]) {
				dfs(n);
			}
		}
	}
}